scripts/setup_irods.py

```diff
@@ -62,6 +62,7 @@ def wrap_if_necessary():
 import irods.log
 from irods.password_obfuscation import maximum_password_length
 from irods.logging_infrastructure import setup_rsyslog_and_logrotate, rsyslog_config_path, logrotate_config_path
+from irods import database_interface
```

Comment:
so a setup run that is not a provider does not need the database_interface - which is why we didn't have it up top...

agreed that earlier is better - but can we preserve the 'only if needed' behavior?   not sure how, since we're pivoting on one of their answers...

Reply:
Perhaps a try-except which prints the warning here (along with a note saying that it may not be required) and then actually raises the exception at the point(s) where it is needed? The import should be attempted again if it happens further down.

I would prefer if this happened on an "only if needed" basis as well.

Is there a reason to do this outside of the current implementation being annoying?

Reply:
Not that I'm aware.
I'll wrap this in a try-except.